rfcs/12-primitive-shapes.md

```diff
@@ -0,0 +1,538 @@
+  circumcircle: Circle2d,
+  /// Number of faces.
+  faces: u8,
+  /// Clockwise rotation of the polygon about the origin. At zero rotation, a point will always be located at the 12 o'clock position.
+  orientation: Angle,
+}
+impl Meshable for RegularPolygon2d {}
+impl Collider2d for RegularPolygon2d {}
+ 
+struct Polygon2d <const N: usize>{
+  points: [Point; N],
+}
+impl Meshable for Polygon2d {}
+impl Collider2d for Polygon2d {}
+
+/// Circle types
+
+struct Circle2d {
+  radius: f32,
+}
+impl Meshable for Circle2d {}
+
+struct BoundingCircle2d {
+  circle: Circle2d,
+  translation: Vec2,
+}
+impl Meshable for BoundingCircle2d {}
+impl Bounding2d for BoundingCircle2d {}
+
+struct CircleCollider2d {
+  sphere: Circle2d,
+  translation: Vec2,
+}
+impl Meshable for CircleCollider2d {}
+impl Collider2d for CircleCollider2d {}
+
+// Box Types
+
+struct Box2d {
+  half_extents: Vec2,
+}
+impl Meshable for Box2d
+
+struct BoundingBox2d {
+  box: Box2d,
+  translation: Vec2,
+}
+impl Meshable for BoundingBox2d {}
+impl Bounding2d for BoundingBox2d {}
+type Aabb2d = BoundingBox2d;
+
+struct BoxCollider2d {
+  box: Box2d,
+  translation: Vec2,
+  rotation: Mat2,
+}
+impl Meshable for BoxCollider2d {}
+impl Collider2d for BoxCollider2d {}
+type Obb2d = BoxCollider2d;
+
+// Capsule Types
+
+struct Capsule2d {
+  // Height of the cylindrical section
+  height: f32,
+  radius: f32,
+}
+impl Meshable for Capsule2d {}
+
+struct CapsuleCollider2d {
+  capsule: Capsule2d,
+  translation: Vec2,
+  rotation: Mat2,
+}
+impl Meshable for CapsuleCollider2d {}
+impl Collider2d for CapsuleCollider2d {}
+
+```
+### Lack of `Transform`s
+
+Primitives colliders and bounding volumes are fully defined in space, and do not use `Transform` or `GlobalTransform`. This is an intentional decision. Because transforms can include nonuniform scale, they are fundamentally incompatible with shape primitives. We could use transforms in the future if the `translation`, `rotation`, and `scale` fields were distinct components, and shape primitives could be bundled with a `translation` or `rotation` if applicable.
+
+#### Cache Efficiency
+
+- Some primitives such as AABB and Sphere don't need a rotation (or scale) to be fully defined. 
+- Using a `GlobalTransform` adds an unused Quat and Vec3 to the cache line.
+- This is especially important for AABBs and Spheres, because they are fundamental to broad phase collision detection and BV(H), and as such need to be as efficient as possible.
+- This also applies to all other primitives, which don't use the `scale` field of the `GlobalTransform`.
+
+#### Ergonomics
+
+- Primitives need to be fully defined in world space to compute collision or bounding.
+- By making the primitive components fully defined and standalone, computing operations is as simple as: `primitive1.some_function(primitive_2)`, instead of also having query and pass in 2 `GlobalTransform`s in the (hopefully) correct order.
```

Comment:
The downside being the need for another system to keep them updated. Which also adds another system ordering constraint.

Reply:
This can be added directly to the transform propagation system, or immediately after in another system if desirable.